crates/uv/src/commands/python/pin.rs

```diff
@@ -33,6 +35,25 @@ pub(crate) async fn pin(
         // Display the current pinned Python version
         if let Some(pins) = requests_from_version_file().await? {
             for pin in pins {
+                let python = match PythonInstallation::find(
+                    &pin,
+                    EnvironmentPreference::OnlySystem,
+                    python_preference,
+                    cache,
+                ) {
+                    Ok(python) => Some(python),
+                    // If no matching Python version is found, don't fail unless `resolved` was requested
+                    Err(uv_python::Error::MissingPython(err)) if !resolved => {
+                        warn_user_once!("{}", err);
+                        None
+                    }
+                    Err(err) => return Err(err.into()),
+                };
+                if !isolated {
+                    if let Some(ref python) = python {
+                        assert_python_compatibility(python).await?;
```

Comment:
This will:

1. Perform workspace lookup for each pin the file
2. Throw an error on the first incompatible pin

Instead, we should:

1. Find the `VirtualProject` one time before doing anything
2. Display warnings for incompatibilities on stderr without failing the display since we're just reading pins here

Reply:
Done